src/api/users.ts

```diff
@@ -0,0 +1,45 @@
+import express from "express";
+import { PrismaClient } from "@prisma/client";
+const bcrypt = require("bcrypt")
+
+const SALT_ROUNDS = 10;
+
+const prisma = new PrismaClient();
+
+const usersRouter = express.Router();
+
+// GET /api/users
+
+usersRouter.get("/", async (req, res, next): Promise<void> => {
+    try {
+        const users = await prisma.user.findMany();
+        res.send({users});
+    } catch (e) {
+        next(e);
+    }
+})
+
+// POST /api/users/register
+
+usersRouter.post("/register", async (req, res, next) => {
+    try {
+        const { email, username, password } = req.body;
+        bcrypt.hash(password, SALT_ROUNDS, async function(err: Error | undefined, hash: string) {
+            const user = await prisma.user.create({
+            data: {
+                email,
+                username,
+                password: hash
+            }
+        })
+        res.send({
+            email: user.email,
+            username: user.username
+        });
```

Comment:
Response should also send a JWT token to use for auth
```suggestion
        res.send({
          token,
          user: {
            email: user.email,
            username: user.username
          }
        });
```

Reply:
Or at least add a TODO to add the token